### PR TITLE
Fix Capstone plugin

### PR DESCRIPTION
**Pull request by hugsy**
## How Has This Been Tested ?

"Tested" indicates that the PR works *and* the unit test (i.e. `make test`) run passes without issue.

*  [ ] x86-32
*  [x] x86-64
*  [ ] ARM
*  [ ] AARCH64
*  [ ] MIPS
*  [ ] POWERPC
*  [ ] SPARC
*  [ ] RISC-V

## Checklist

*  [ ] My code follows the code style of this project.
*  [ ] My change includes a change to the documentation, if required.
*  [ ] If my change adds new code,
   [adequate tests](https://hugsy.github.io/gef/testing) have been added.
*  [ ] I have read and agree to the
   [CONTRIBUTING](https://github.com/hugsy/gef/blob/main/.github/CONTRIBUTING.md) document.

### CODE REVIEW

**Review by therealdreg — CHANGES_REQUESTED** [review]
The documentation refers to the alias "**cs**," but the alias being registered in the code is: "**cs-dis**"

image

image

File: scripts/capstone.py [hunk 1]
```diff
@@ -182,7 +182,7 @@ def do_invoke(self, _: List[str], **kwargs: Any) -> None:
         args = kwargs["arguments"]
         show_opcodes = args.show_opcodes
         length = args.length or gef.config["context.nb_lines_code"]
-        location = parse_address(args.location)
+        location = int(gdb.parse_and_eval(args.location).address)
```
Comment (therealdreg) [inline scripts/capstone.py:185]:
```suggestion
        location = location = int(parse_address(args.location))
```

**int(gdb.parse_and_eval(args.location).address)** is not working

**Review by therealdreg — CHANGES_REQUESTED** [review]
The value from the configuration file is not taken into account.

So, capstone-disassemble.use-capstone is always False (on every boot)

here my conf file:
```
[capstone-disassemble]
use-capstone = True
```

```
gef➤  gef config capstone-disassemble.use-capstone 
──────────────────────────────────────── GEF configuration setting: capstone-disassemble.use-capstone ────────────────────────────────────────
capstone-disassemble.use-capstone (bool) = False

Description:
	Replace the GDB disassembler in the `context` with Capstone
gef➤  
```  

@hugsy **Is it possible that this issue is happening with more plugins?**

**Comment by hugsy** [comment]
> The value from the configuration file is not taken into account.
> 
> So, capstone-disassemble.use-capstone is always False (on every boot)
> 
> here my conf file:
> 
> ```
> [capstone-disassemble]
> use-capstone = True
> ```
> 
> ```
> gef➤  gef config capstone-disassemble.use-capstone 
> ──────────────────────────────────────── GEF configuration setting: capstone-disassemble.use-capstone ────────────────────────────────────────
> capstone-disassemble.use-capstone (bool) = False
> 
> Description:
> 	Replace the GDB disassembler in the `context` with Capstone
> gef➤  
> ```
> 
> @hugsy **Is it possible that this issue is happening with more plugins?**

This is what I explained to you and partially fixed in #1090

image

All gef-extras settings are impacted

**Comment by therealdreg** [comment]
> > The value from the configuration file is not taken into account.
> > So, capstone-disassemble.use-capstone is always False (on every boot)
> > here my conf file:
> > ```
> > [capstone-disassemble]
> > use-capstone = True
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > ```
> > gef➤  gef config capstone-disassemble.use-capstone 
> > ──────────────────────────────────────── GEF configuration setting: capstone-disassemble.use-capstone ────────────────────────────────────────
> > capstone-disassemble.use-capstone (bool) = False
> > 
> > Description:
> > 	Replace the GDB disassembler in the `context` with Capstone
> > gef➤  
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > @hugsy **Is it possible that this issue is happening with more plugins?**
> 
> This is what I explained to you and partially fixed in #1090
> 
> image
> 
> All gef-extras settings are impacted

And what can we do to fix this? I wouldn’t like to release the next version with this issue...

**Comment by hugsy** [comment]
> And what can we do to fix this?

`gef restore`

But I think this should not happen with the latest change to gef because now loading gef-extras with `on_changed` callback should result in a config reloading.

**Comment by therealdreg** [comment]
> > And what can we do to fix this?
> 
> `gef restore`
> 
> But I think this should not happen with the latest change to gef because now loading gef-extras with `on_changed` callback should result in a config reloading.

I am using the last version (GEF+EXTRAS) and the problem is still here :\

**Comment by therealdreg** [comment]
The command works fine, and the alias is fixed!

[image](https://github.com/hugsy/gef-extras/assets/9882181/71a542ba-7245-4264-abf2-bea585e28d6a)

But on each run capstone is set to false!

image

image

**Review by therealdreg — CHANGES_REQUESTED** [review]
on each run capstone is set to false!

----

dreg@rootkit:~/.config/gef-extras$ git log
commit 0b7041fec0a7a5fcf6913676585f7473a058879d (HEAD -> fix_capstone_disass_switch, origin/fix_capstone_disass_switch)
Author: hugsy <crazy.hugsy@outlook.com>
Date:   Mon Jun 3 09:51:40 2024 -0700

    make dreg happy

commit 296049f7d199a4375267cc3d0db62fb30a420cef (tag: 2024.06, origin/main, origin/HEAD, main)
Author: MinatoTW <shaks19jais@gmail.com>
Date:   Thu May 16 08:39:56 2024 +0530

    64bit libc table generator had $r10 as 4th argument (#113)
    
    ## Description/Motivation/Screenshots
    
    Fixes `scripts/libc_function_args/tables/generator.py` which was wrong using `$r10` as 4th argument

----

GEF: (Standalone)
Blob Hash(/home/dreg/.gef-2024.06.py): 88981e223320723f9df39bd8714ea2d56da4dbee
SHA256(/home/dreg/.gef-2024.06.py): 764738509912bea65f67927691d5fa0421444d1969678208095733fdbf0dd83d
GDB: 15.0.50.20240403-git
GDB-Python: 3.12

**Comment by hugsy** [comment]
This is a different problem to what this PR fixes (and again it can be fixed by doing `gef restore` in your gdbinit)

**Review by therealdreg — APPROVED** [review]

**Comment by therealdreg** [comment]
> This is a different problem to what this PR fixes (and again it can be worked around by doing `gef restore` in your gdbinit) In GEF you can do a PR that will execute `gef restore` if `load_plugins_from_directory()` return > 0
> 
> Different problems.

Okay, we'll fix it in another PR. Thanks for your patience :D

**Comment by therealdreg** [comment]
https://github.com/hugsy/gef/pull/1118 fixes the problem!  thx!

image

**Pull request merged**